addons/web/static/src/webclient/commands/command_service.js

```diff
@@ -38,40 +52,20 @@ export const commandService = {
                 return;
             }
 
-            const overlayModifier = registry.category("services").get("hotkey").overlayModifier;
-            const commands = [...registeredCommands.values()].filter(
-                (command) => command.activeElement === ui.activeElement
-            );
-
-            // Also retrieve all hotkeyables elements
-            for (const el of ui.getVisibleElements("[data-hotkey]:not(:disabled)")) {
-                const closest = el.closest("[data-command-category]");
-                const category = closest ? closest.dataset.commandCategory : "default";
-
-                const description =
-                    el.title ||
-                    el.dataset.originalTitle || // LEGACY: bootstrap moves title to data-original-title
-                    el.placeholder ||
-                    (el.innerText &&
-                        `${el.innerText.slice(0, 50)}${el.innerText.length > 50 ? "..." : ""}`) ||
-                    "no description provided";
-                commands.push({
-                    name: capitalize(description.trim().toLowerCase()),
-                    hotkey: `${overlayModifier}+${el.dataset.hotkey}`,
-                    action: () => {
-                        // AAB: not sure it is enough, we might need to trigger all events that occur when you actually click
-                        el.focus();
-                        el.click();
-                    },
-                    category,
-                });
-            }
+            const commandProviders = [
+                // ...registeredProviders.values(), TODO TO Discuss
+                ...commandProviderRegistry.getAll(),
+            ];
 
             // Open palette dialog
             isPaletteOpened = true;
             dialog.add(
                 CommandPaletteDialog,
-                { commands },
+                {
```

Comment:
i think that these props should not be given to the dialog, except activeelement because the command palette cannot get them itself.

Reply:
or is it true? can't the command palette get the activeelement in its setup method?